lib/logstash/inputs/redis.rb

```diff
@@ -151,18 +154,27 @@ def internal_redis_builder
   # private
   def connect
     redis = new_redis_instance
+
+    # register any renamed Redis commands
+    if @command_map and not @command_map.empty?
+      client_command_map = redis.client.command_map
+      @command_map.each do |name, renamed|
+        client_command_map[name.to_sym] = renamed.to_sym
+      end
+    end
+
     load_batch_script(redis) if batched? && is_list_type?
     redis
   end # def connect
 
   # private
   def load_batch_script(redis)
     #A Redis Lua EVAL script to fetch a count of keys
-      redis_script = <<EOF
-        local batchsize = tonumber(ARGV[1])
-        local result = redis.call('lrange', KEYS[1], 0, batchsize)
-        redis.call('ltrim', KEYS[1], batchsize + 1, -1)
-        return result
+    redis_script = <<EOF
+      local batchsize = tonumber(ARGV[1])    
+      local result = redis.call('#{@command_map['lrange'] || 'lrange'}', KEYS[1], 0, batchsize)
```

Comment:
this call will have an argument that is the raw string `'#{@command_map['lrange'] || 'lrange'}'` because single quotes are being used, I believe this is not the original intent? Example:

```ruby
jruby-9.2.0.0 :001 > a = 1
 => 1 
jruby-9.2.0.0 :002 > '#{a}'
 => "\#{a}" 
jruby-9.2.0.0 :003 > "#{a}"
 => "1" 
```

this can be instead (same for trim):

```ruby
redis.call(@command_map.fetch('lrange', 'lrange'), KEYS[1], 0, batchsize)
```

Reply:
nevermind, this is all wrapped in a literal string, which will perform the interpolation correctly, 🤦‍♂️ 

Still I suggest switching to fetch just for code clarity, and I think we're good to merge